src/App.tsx

```diff
@@ -14,11 +15,98 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[],
+  selectedGoods: string[]
+}
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    selectedGoods: ['Jam'],
+  };
+
+  getButtonAction = (action: string, good?: string) => {
+    if (good && action === 'select') {
+      this.setState((state) => ({ selectedGoods: [...state.selectedGoods, good] }));
+    } else if (good && action === 'remove') {
+      this.setState((state) => ({ selectedGoods: state.selectedGoods.filter(el => el !== good) }));
+    } else {
+      this.setState({ selectedGoods: [] });
+    }
+  };
+
+  render() {
+    const { selectedGoods, goods } = this.state;
+
+    return (
+      <div className="app">
+        <h1 className="title">
+          {selectedGoods.length > 0
+            ? `${selectedGoods.join(', ')} ${selectedGoods.length > 1 ? 'are' : 'is'}`
+              + ' selected'
+            : 'No goods selected'}
+        </h1>
+
+        {selectedGoods.length !== 0
+          && (
+            <button
+              type="button"
+              className="button button--big"
+              onClick={() => {
+                this.getButtonAction('ClearAll');
```

Comment:
do you need to put here something if method does not handle it?
better create a separate method for clearing, it may be more code but it is much easier to maintain and reuse